web/src/components/Market/Outcomes.tsx

```diff
@@ -202,13 +215,57 @@ export function Outcomes({ market, images }: PositionsProps) {
   const { Modal, openModal, closeModal } = useModal("liquidity-modal");
   const blockExplorerUrl = SUPPORTED_CHAINS[market.chainId].blockExplorers?.default?.url;
 
+  const routerAddress = getRouterAddress(market.chainId);
+  const { data: marketStatus } = useMarketStatus(market);
+  const winningOutcomes = useQuery({
+    queryKey: ["winningOutcomes", market.conditionId, routerAddress],
+    enabled: marketStatus === MarketStatus.CLOSED && !!routerAddress,
+    queryFn: async () => {
+      return await readContract(config, {
+        abi: RouterAbi,
+        address: routerAddress as Address,
+        functionName: "getWinningOutcomes",
+        args: [market.conditionId],
+        chainId: market.chainId
+      });
+    }
+  });
+
   const indexesOrderedByOdds = useMemo(() => {
     if (oddsPending || odds.length === 0) {
       return null;
     }
-    const oddsAndIndexes = odds.map((odd, i) => ({ odd, i })).sort((a, b) => b.odd - a.odd);
-    return oddsAndIndexes.map((obj) => obj.i);
-  }, [odds]);
+    const invalidIndex = market.outcomes.findIndex(outcome => outcome === INVALID_RESULT_OUTCOME_TEXT);
```

Comment:
the code inside the useMemo seems to be duplicated on both files, the OutcomeWithOdds type definition too. Let's move all of this to a function to allow to reuse it everywhere it's needed (maybe it can even be a react query hook, your choice :))